### ISSUE TITLE
tf.estimator.train_and_evaluate  "loss" meaning in "saving dict for global step"

### ISSUE THREAD

**Issue by haozaiiii** [tensorflow/tensorflow#43014]
when i run tf.estimator.train_and_evaluate, log prints that:
[INFO:tensorflow:Saving dict for global step 1000: eval_loss = 0.0055662687, global_step = 1000, loss = 3.3700151]
INFO:tensorflow:global_steps = 1000, loss = 0.05528491 (1202.815 sec)

question:
what's meaning of the "loss = 3.3700151"

=====================log=====================
```
2020-09-07 12:33:03.540827: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2020-09-07 12:33:03.542687: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-09-07 12:33:03.542709: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2020-09-07 12:33:03.542718: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2020-09-07 12:33:03.542826: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 22853 MB memory) -> physical GPU (device: 0, name: TITAN RTX, pci bus id: 0000:09:00.0, compute capability: 7.5)
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 0 into /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt.
2020-09-07 12:33:26.224454: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
INFO:tensorflow:loss = 96.96749, step = 0
INFO:tensorflow:global_steps = 0, loss = 96.96749
INFO:tensorflow:global_step/sec: 0.803899
INFO:tensorflow:loss = 4.264659, step = 100 (124.394 sec)
INFO:tensorflow:global_step/sec: 0.682575
INFO:tensorflow:loss = 3.211001, step = 200 (146.504 sec)
INFO:tensorflow:global_step/sec: 0.442842
INFO:tensorflow:loss = 0.8785575, step = 300 (225.815 sec)
INFO:tensorflow:global_step/sec: 0.388806
INFO:tensorflow:loss = 1.20301, step = 400 (257.197 sec)
INFO:tensorflow:global_step/sec: 0.389973
INFO:tensorflow:loss = 0.50117, step = 500 (256.429 sec)
INFO:tensorflow:global_steps = 500, loss = 0.50117 (1010.338 sec)
INFO:tensorflow:global_step/sec: 0.431264
INFO:tensorflow:loss = 0.52161276, step = 600 (231.876 sec)
INFO:tensorflow:global_step/sec: 0.42097
INFO:tensorflow:loss = 0.32268906, step = 700 (237.547 sec)
INFO:tensorflow:global_step/sec: 0.409772
INFO:tensorflow:loss = 0.28711706, step = 800 (244.038 sec)
INFO:tensorflow:global_step/sec: 0.398877
INFO:tensorflow:loss = 0.32013908, step = 900 (250.704 sec)
INFO:tensorflow:Saving checkpoints for 1000 into /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt.
INFO:tensorflow:Calling model_fn.
I:NER Training:[ber:mod:384]:*** Features ***
I:NER Training:[ber:mod:386]:  name = input_ids, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = input_mask, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = label_ids, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = segment_ids, shape = (?, 202)
WARNING:tensorflow:From /home/panyinghao/anaconda3/envs/tf13/lib/python3.6/site-packages/tensorflow/python/ops/metrics_impl.py:363: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2020-09-07T05:10:03Z
INFO:tensorflow:Graph was finalized.
2020-09-07 13:10:03.661048: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2020-09-07 13:10:03.661119: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-09-07 13:10:03.661130: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2020-09-07 13:10:03.661137: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2020-09-07 13:10:03.661247: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 22853 MB memory) -> physical GPU (device: 0, name: TITAN RTX, pci bus id: 0000:09:00.0, compute capability: 7.5)
WARNING:tensorflow:From /home/panyinghao/anaconda3/envs/tf13/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
INFO:tensorflow:Restoring parameters from /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt-1000
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Finished evaluation at 2020-09-07-05:10:18
INFO:tensorflow:Saving dict for global step 1000: eval_loss = 0.0055662687, global_step = 1000, loss = 3.3700151
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 1000: /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt-1000
INFO:tensorflow:global_step/sec: 0.419043
WARNING:tensorflow:From /home/panyinghao/anaconda3/envs/tf13/lib/python3.6/site-packages/tensorflow/python/summary/summary_iterator.py:68: tf_record_iterator (from tensorflow.python.lib.io.tf_record) is deprecated and will be removed in a future version.
Instructions for updating:
Use eager execution and: 
`tf.data.TFRecordDataset(path)`
INFO:tensorflow:loss = 0.05528491, step = 1000 (238.650 sec)
INFO:tensorflow:global_steps = 1000, loss = 0.05528491 (1202.815 sec)
INFO:tensorflow:global_step/sec: 0.66271
INFO:tensorflow:loss = 0.20649494, step = 1100 (150.885 sec)
INFO:tensorflow:global_step/sec: 0.681622
INFO:tensorflow:loss = 0.21718065, step = 1200 (146.709 sec)
INFO:tensorflow:global_step/sec: 0.684914
INFO:tensorflow:loss = 0.12876259, step = 1300 (146.004 sec)
INFO:tensorflow:global_step/sec: 0.680007
INFO:tensorflow:loss = 0.100298606, step = 1400 (147.057 sec)
INFO:tensorflow:global_step/sec: 0.676314
INFO:tensorflow:loss = 0.12061991, step = 1500 (147.873 sec)
INFO:tensorflow:global_steps = 1500, loss = 0.12061991 (738.528 sec)
INFO:tensorflow:global_step/sec: 0.642209
INFO:tensorflow:loss = 0.0042267526, step = 1600 (155.699 sec)
INFO:tensorflow:global_step/sec: 0.637592
INFO:tensorflow:loss = 0.0035796848, step = 1700 (156.841 sec)
INFO:tensorflow:Saving checkpoints for 1793 into /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt.
INFO:tensorflow:Calling model_fn.
I:NER Training:[ber:mod:384]:*** Features ***
I:NER Training:[ber:mod:386]:  name = input_ids, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = input_mask, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = label_ids, shape = (?, 202)
I:NER Training:[ber:mod:386]:  name = segment_ids, shape = (?, 202)
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2020-09-07T05:30:20Z
INFO:tensorflow:Graph was finalized.
2020-09-07 13:30:21.334671: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2020-09-07 13:30:21.334728: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-09-07 13:30:21.334737: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2020-09-07 13:30:21.334743: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2020-09-07 13:30:21.835343: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 22853 MB memory) -> physical GPU (device: 0, name: TITAN RTX, pci bus id: 0000:09:00.0, compute capability: 7.5)
INFO:tensorflow:Restoring parameters from /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt-1793
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Finished evaluation at 2020-09-07-05:30:38
INFO:tensorflow:Saving dict for global step 1793: eval_loss = 0.005485246, global_step = 1793, loss = 4.2882214
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 1793: /home/panyinghao/model/checkpoints/test/test_10ep/model.ckpt-1793
INFO:tensorflow:Loss for final step: 0.12153285.
```

**Comment by Saduf2019** [comment]
@haozaiiii 
Please fill in issue template for us to replicate the issue faced.

**Comment by haozaiiii** [comment]
ok

**Comment by haozaiiii** [comment]
> @haozaiiii
> Please fill in issue template for us to replicate the issue faced.

ok

**Comment by Saduf2019** [comment]
@haozaiiii 
This question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
Please feel free to close the issue if you do not have any other query.
